pkg/util/network.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/ec2"
+	"github.com/DataDog/datadog-agent/pkg/util/gce"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GetNetworkID retrieves the network_id which can be used to improve network
+// connection resolution. This can be configured or detected.  The
+// following sources will be queried:
+// * configuration
+// * GCE
+// * EC2
+func GetNetworkID() (string, error) {
+	cacheNetworkIDKey := cache.BuildAgentKey("networkID")
+	if cacheNetworkID, found := cache.Cache.Get(cacheNetworkIDKey); found {
+		return cacheNetworkID.(string), nil
+	}
+
+	// the the id from configuration
+	if networkID := config.Datadog.GetString("network.id"); networkID != "" {
+		cache.Cache.Set(cacheNetworkIDKey, networkID, cache.NoExpiration)
+		return networkID, nil
+	}
+
+	log.Debugf("GetNetworkID trying GCE")
+	if networkID, err := gce.GetNetworkID(); err != nil {
+		cache.Cache.Set(cacheNetworkIDKey, networkID, cache.NoExpiration)
+		return networkID, nil
```

Comment:
Can we add another debug log line saying the the networkID was found with GCE (and even logging the networkID if it's not a sensitive info) ?

Reply:
Yes, done.